src/pytest_replay/__init__.py

```diff
@@ -88,16 +89,19 @@ def pytest_runtest_logstart(self, nodeid):
     def pytest_runtest_makereport(self, item):
         report = yield
         result = report.get_result()
-        if self.dir and result.when == "teardown":
-            json_content = json.dumps(
-                {
-                    "nodeid": item.nodeid,
-                    "start": self.node_start_time[item.nodeid],
-                    "finish": time.perf_counter() - self.session_start_time,
-                    "outcome": result.outcome,
-                }
-            )
-            self.append_test_to_script(item.nodeid, json_content)
+        if self.dir:
+            if result.when == "call":
+                self.node_outcome[item.nodeid] = result.outcome
+            elif result.when == "teardown":
```

Comment:
```suggestion
            self.node_outcome.setdefault(item.nodeid, result.outcome)
            if result.failed:
                self.node_outcome[item.nodeid] = result.outcome
            if result.when == "teardown":
```
This way all phases' outcomes are considered or a test could be flagged as "passed" even when a fixture teardown failed.

Reply:
```suggestion
            self.node_outcome.setdefault(item.nodeid, result.outcome)
            if not result.passed:
                self.node_outcome[item.nodeid] = result.outcome
            if result.when == "teardown":
```
BTW, "if not passed" is probably better. Given there is 3 possible outcomes.

Reply:
this would not account for this case though:

```py
import pytest

@pytest.fixture()
def foo():
    yield
    pytest.skip("skipping")

def test(foo):
    assert False
```

This would overwrite "failed" with "skipped", while I guess the test should be considered failed.